shm/cargo.cpp

```diff
@@ -0,0 +1,37 @@
+#include "cargo.hpp"
+
+#include <iostream>
+
+Cargo::Cargo(uint32_t amount, std::string name, uint32_t basePrice)
+    : amount_(amount), name_(name), basePrice_(basePrice) {}
+Cargo::~Cargo() {}
+
+std::string Cargo::getName() const {
+    return name_;
+}
+
+uint32_t Cargo::getAmount() const {
+    return amount_;
+}
+
+uint32_t Cargo::getBasePrice() const {
+    return basePrice_;
+}
+
+Cargo& Cargo::operator+=(uint32_t amount) {
+    amount_ += amount;
+    return *this;
+}
+
+Cargo& Cargo::operator-=(uint32_t amount) {
+    if (amount <= amount_) {
+        amount_ -= amount;
+        return *this;
+    }
+    std::cout << "There is no so many cargo!";
```

Comment:
Maybe std::cerr? This situation should never happen, because you should always check if you can decrease amount. So std::cerr will be better to print error.